frontend/src/components/note/NoteModal.vue

```diff
@@ -71,108 +74,71 @@ const formSchema = object({
-          formRef.value ? formRef.value.resetForm() : null;
-        }
+          })
+          .catch((e: any) => toast.error('Failed to delete note', e.message))
+          .finally(() => {
+            visible.value = false;
+            formRef.value?.resetForm();
+          });
       }
     });
   }
-};
-
-const setShowBringForwardState = () => {
-  formRef.value?.setFieldValue('bringForwardState', BRING_FORWARD_TYPES.UNRESOLVED);
-  showBringForward.value = true;
-};
-
-const setHideBringForwardState = () => {
-  showBringForward.value = false;
-  formRef.value?.setFieldValue('bringForwardDate', null);
-  formRef.value?.setFieldValue('bringForwardState', null);
-};
+}
 
-const handleBringForward = (e: { OriginalEvent: Event; value: string }) => {
+const onNoteTypeChange = (e: { OriginalEvent: Event; value: string }) => {
   if (e.value === NOTE_TYPES.BRING_FORWARD) {
-    setShowBringForwardState();
+    formRef.value?.setFieldValue('bringForwardState', BRING_FORWARD_TYPES.UNRESOLVED);
+    showBringForward.value = true;
   } else {
-    setHideBringForwardState();
+    formRef.value?.setFieldValue('bringForwardDate', null);
+    formRef.value?.setFieldValue('bringForwardState', null);
+    showBringForward.value = false;
   }
 };
 
-watch(visible, (newValue) => {
-  // nextTick triggers when modal becomes visible
-  nextTick().then(() => {
-    if (!newValue) {
-      setHideBringForwardState();
-      return;
-    }
-
-    // sets 'createdAt' to current time when modal is visible
-    if (formRef.value) {
-      formRef.value.setFieldValue('createdAt', new Date());
-
-      // set form values if editing existing note
-      if (props.note) {
-        formRef.value.setFieldValue('noteId', props.note.noteId);
-        formRef.value.setFieldValue('activityId', props.note.activityId);
-        formRef.value.setFieldValue('bringForwardState', props.note.bringForwardState);
-        formRef.value.setFieldValue('noteType', props.note.noteType);
-        if (props.note.noteType === NOTE_TYPES.BRING_FORWARD) {
-          setShowBringForwardState();
-          formRef.value.setFieldValue(
-            'bringForwardDate',
-            props.note.bringForwardDate ? new Date(props.note.bringForwardDate) : null
-          );
-        }
-        formRef.value.setFieldValue('note', props.note.note);
-        formRef.value.setFieldValue('title', props.note.title);
-      }
+// @ts-expect-error TS7031
+// resetForm is an automatic binding https://vee-validate.logaretm.com/v4/guide/components/handling-forms/
+async function onSubmit(data: any, { resetForm }) {
+  try {
+    if (!props.note) {
+      const result = (await noteService.createNote({ ...data, activityId: props.activityId })).data;
+      submissionStore.addNote(result, true);
+      resetForm();
+    } else {
+      const result = (await noteService.updateNote({ ...data, activityId: props.activityId })).data;
+      submissionStore.updateNote(props.note, result);
```

Comment:
When a note is updated with a new date, it remains in place sorted in the frontend with the old date's value until the page is refreshed.